From 00502f3114955702c783fd0edc850ffa49320fa8 Mon Sep 17 00:00:00 2001
From: hienhuynh <hien.huynh.px@renesas.com>
Date: Mon, 20 Jun 2022 13:27:30 +0700
Subject: [PATCH 01/27] pinctrl: pinctrl-rzg2l: correct shift bit for TITSR0/1

This commit is created to correct shift bit for TITSR0/1.
- TITSR0: tint_slot should be from 0 to 15.
- TITSR1: tint_slot should be from 16 to 31.

Signed-off-by: hienhuynh <hien.huynh.px@renesas.com>
---
 drivers/pinctrl/renesas/pinctrl-rzg2l.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/pinctrl/renesas/pinctrl-rzg2l.c b/drivers/pinctrl/renesas/pinctrl-rzg2l.c
index a3b79f659b80..4c911b5942f6 100644
--- a/drivers/pinctrl/renesas/pinctrl-rzg2l.c
+++ b/drivers/pinctrl/renesas/pinctrl-rzg2l.c
@@ -1130,13 +1130,13 @@ static int rzg2l_gpio_irq_set_type(struct irq_data *d, unsigned int type)
 
 	if (tint_slot > 15) {
 		reg32 = readl(pctrl->base_tint + TITSR1);
-		reg32 &= ~(IRQ_MASK << (tint_slot * 2));
-		reg32 |= irq_type << (tint_slot * 2);
+		reg32 &= ~(IRQ_MASK << ((tint_slot - 16) * 2));
+		reg32 |= irq_type << ((tint_slot - 16) * 2);
 		writel(reg32, pctrl->base_tint + TITSR1);
 	} else {
 		reg32 = readl(pctrl->base_tint + TITSR0);
-		reg32 &= ~(IRQ_MASK << ((tint_slot - 16) * 2));
-		reg32 |= irq_type << ((tint_slot - 16) * 2);
+		reg32 &= ~(IRQ_MASK << (tint_slot * 2));
+		reg32 |= irq_type << (tint_slot * 2);
 		writel(reg32, pctrl->base_tint + TITSR0);
 	}
 
-- 
2.17.1

